src/shared/mkdir-label.c

```diff
@@ -57,3 +57,11 @@ int mkdir_parents_label(const char *path, mode_t mode) {
 int mkdir_p_label(const char *path, mode_t mode) {
         return mkdir_p_internal(NULL, path, mode, UID_INVALID, UID_INVALID, 0, mkdir_label);
 }
+
+void mkdir_p_label_and_warn(const char *path, mode_t mode, const char *logsrc) {
+        int r;
+
+        r = mkdir_p_label(path, mode);
+        if (r < 0 && r != -EEXIST)
+                log_warning_errno(r, "%s%sFailed to create dir '%s', ignoring: %m", logsrc ?: "", logsrc ? ": " : "", path);
```

Comment:
Please do not always log higher than LOG_DEBUG. So, please use `log_debug_errno()`. Or, make this function take one more argument `int log_level`, then use `log_full_errno()`, then only specific place use a higher log level.

Also, do not eat the error, but propagate to the caller. Even if it is used nowhare.

Also, 
`logsrc ?: ""` -> `strempty(logsrc)`
`logsrc ? ": " : ""` -> `isempty(logsrc) ? "" : ": "`

Reply:
> Please do not always log higher than LOG_DEBUG. So, please use `log_debug_errno()`. Or, make this function take one more argument `int log_level`, then use `log_full_errno()`, then only specific place use a higher log level.
>
> Also, do not eat the error, but propagate to the caller. Even if it is used nowhare.

I initially did do this (see previous push), but... the way I think of it, this is a function optimized for the "try to mkdir, warn if it fails, and continue" case described by @poettering. If you don't want to warn, or you don't want to ignore errors, then you can use the regular function. It seems like trying to make this a general-purpose API defeats the purpose a bit of having a tailor-made ergonomic function like this with clear semantics.

> Also, `logsrc ?: ""` -> `strempty(logsrc)` `logsrc ? ": " : ""` -> `isempty(logsrc) ? "" : ": "`

Ack, will change.

Reply:
lots of calls were replaced though, are they all related to automount?

Reply:
No, they were all the instance of `mkdir_p_label` I could find where we ignored errors. I can scale this down to just the few that brought me here. There might be some callsites where we really don't want to warn though I'm thinking those are likely the exception.

Reply:
given the log level is warning, I'd prefer not to risk spamming the journal with messages that are perfectly fine to ignore - your use case is of course one where we should log, but unless they were all analyzed and it turns out they should log, I'd leave them be